agent/engine/docker_task_engine.go

```diff
@@ -625,7 +654,19 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
+	if dockerContainerMD.Error == nil && engine.cfg.ContainerMetadataEnabled == true {
+		err := engine.metadataManager.UpdateMetadata(dockerContainer.DockerID, task, container)
+		if err != nil {
+			seelog.Errorf("Failed to update metadata file for container %s of task %s: %v", container, task, err)
```

Comment:
First, I would like to make clear that "dockerContainerMD" is unrelated to the container metadata service and is only used to check if start container was successful. Second, since the error is consumed here and only used for logging, I do not see how it could cancel the container launch.